src/Build/Evaluation/Expander.cs

```diff
@@ -4534,6 +4538,29 @@ private static bool TryConvertToInt(object value, out int arg0)
                 return false;
             }
 
+            private static bool TryConvertToLong(object value, out long arg0)
+            {
+                switch (value)
+                {
+                    case double d:
+                        if (d >= long.MinValue && d <= long.MaxValue)
+                        {
+                            arg0 = Convert.ToInt64(d);
```

Comment:
Convert will throw for `Convert.ToInt64((double)long.MaxValue)` (credit to @rokonec for discovering this)

Please use `long.CreateTruncating(d)` instead - as it grace handles those boundaries.
Same for the Int32 above.

Reply:
Or reuse the code from runtime - so that it works for fullFW as well:

```
arg0 = (d >= long.MaxValue) ? long.MaxValue : (d <= long.MinValue) ? long.MinValue : (long)d;
```

Reply:
The two functions should not throw but also should return false instead of truncating when a `double` can't be narrowed.

Reply:
They should not (would `Convert` behave to its specification), but unfortunately they do...

```csharp
	try
	{
		TryConvertToLong((double)long.MaxValue, out _);
	}
	catch(OverflowException)
	{
		Console.WriteLine("This should never happen!! Or it might? ;-) ...");
	}
```

Btw. we should add a test that exercise that (then add the fixing check for boundary values). Extracting + exposing functions for easier testability is fine.

Reply:
> Convert will throw for Convert.ToInt64((double)long.MaxValue) (credit to @rokonec for discovering this)

The above quote may have been simplifying but `Convert.ToInt64(double)` will throw an OverflowException for values >= (long.MaxValue - 511) and <= long.MaxValue.

Reply:
New commit for unit tests and changes to TryConvertToInt, TryConvertToLong, TryConvertToDouble functions.